classes/plugins/ThemePlugin.inc.php

```diff
@@ -28,107 +51,358 @@ function ThemePlugin() {
+	 * theme.
+	 *
+	 * @return null
+	 */
+	public abstract function init();
+
+	/**
+	* Determine whether or not this plugin is currently active
+	*
+	* This only returns true if the theme is currently the selected theme
+	* in a given context. Use self::getEnabled() if you want to know if the
+	* theme is available for use on the site.
+	*
+	* @return boolean
+	*/
+	public function isActive() {
+		$request = $this->getRequest();
+		$context = $request->getContext();
+		if (is_a($context, 'Context')) {
+			$activeTheme = $context->getSetting('themePluginPath');
+		} else {
 			$site = $request->getSite();
-			$contextOrSite = $context?$context:$site;
+			$activeTheme = $site->getSetting('themePluginPath');
+		}
 
-			// Add the stylesheet.
-			if ($contextOrSite->getSetting('themePluginPath') == basename($path)) {
-				$dispatcher = $request->getDispatcher();
-				$templateManager->addStyleSheet($dispatcher->url($request, ROUTE_COMPONENT, null, 'page.PageHandler', 'css', null, array('name' => $this->getName())), STYLE_SEQUENCE_LATE);
+		return $activeTheme == basename($this->getPluginPath());
+	}
 
-				// If this theme uses templates, ensure they're given priority.
-				array_unshift(
-					$templateManager->template_dir,
-					$path = Core::getBaseDir() . DIRECTORY_SEPARATOR . $this->getPluginPath() . '/templates'
-				);
+	/**
+	 * Add a stylesheet to load with this theme
+	 *
+	 * Style paths with a .less extension will be compiled and redirected to
+	 * the compiled file.
+	 *
+	 * @param string $name A name for this stylesheet
+	 * @param string $path The path to this stylesheet, relative to the theme
+	 * @param array $args Optional arguments hash. Supported args:
+	 *   'context': Whether to load this on the `frontend` or `backend`.
+	 *      default: `frontend`
+	 *   'priority': Controls order in which styles are printed
+	 *   'addLess': Additional LESS files to process before compiling. Array
+	 */
+	public function addStyle($name, $path, $args = array()) {
+
+		// Pass a file path for LESS files
+		if (substr($path, -4) == 'less' ) {
+			$fullPath = $this->_getBaseDir($path);
+
+		// Pass a URL for other files
+		} else {
+			if (isset($args['baseUrl'])) {
+				$fullPath = $args['baseUrl'] . $path;
+			} else {
+				$fullPath = $this->_getBaseUrl($path);
 			}
 		}
-		return true;
+
+		// Generate file paths for any additional LESS files to compile with
+		// this style
+		if (isset($args['addLess'])) {
+			foreach ($args['addLess'] as &$file) {
+				$file = $this->_getBaseDir($file);
+			}
+		}
+
+		$this->styles[$name] = array(
+			'path' => $fullPath,
+			'context' => isset($args['context']) && $args['context'] == 'backend' ? 'backend' : 'frontend',
+			'priority' => isset($args['priority']) ? (int) $args['priority'] : STYLE_SEQUENCE_NORMAL,
+			'addLess' => isset($args['addLess']) ? (array) $args['addLess'] : array(),
+			'baseUrl' => isset($args['baseUrl']) ? (string) $args['baseUrl'] : '',
```

Comment:
I like your approach but I'm not sure if it's best in this case  because I'm also performing some type casting.

If I do it like this, I still have to perform the `isset()` checks and cast the values to `int`, `array`, or `string` separately:

```
		$this->styles[$name] = array_merge(
			array(
				'path' => $fullPath,
				'context' => 'frontend',
				'priority' => STYLE_SEQUENCE_NORMAL,
				'addLess' => array(),
				'baseUrl' => '',
			),
			$args
		);
```

Happy to do so but seems just a little bit more verbose to me. But if it's easier to read I'm happy to go that route.